src/rf/js/src/home/components/library.js

```diff
@@ -356,7 +363,7 @@ var LayerItem = React.createBackboneClass({
     },
 
     deleteLayer: function() {
-        console.log('delete layer');
+        this.props.deleteLayer(this.getModel());
```

Comment:
I think you could just call `this.getModel().destroy()` and all the deletion and removing the item from the collection would happen automatically.